don't transition leader/follower until previous one is done

deepthidevaki opened the pull request:
closes #2355

Zelldon commented on qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/BrokerLeaderChangeTest.java:
```diff
@@ -135,6 +135,50 @@ public void shouldReactToLeaderChanges() {
+    election.async().anoint("1");
+    election.async().anoint("2");
+
+    assertThat(election.getLeadership().leader().id()).isEqualTo(String.valueOf(2));
```
The assertion is a bit flaky and also the assertions below

deepthidevaki commented:
@zelldon I have updated PR as we discussed. Topology is updated only after leader/follower services are installed. In addition, before transitionFuture in PartitionInstallService was completed when the install is complete, now it also waits for the removals. Also moved all the services for leader into a composite service.

deepthidevaki commented:
bors r+

deepthidevaki commented:
bors retry
